### PR TITLE
feat(networking): Add custom DNS for `aws_*` sinks

### PR DESCRIPTION
**Pull request by LucioFranco**
Signed-off-by: Lucio Franco <luciofranco14@gmail.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -93,6 +93,7 @@ bytes = { version = "0.4.10", features = ["serde"] }
 stream-cancel = "0.4.3"
 prometheus = "0.4.2"
 hyper = "0.12.35"
+hyper-openssl = "0.7"
```
Comment (LucioFranco) [inline Cargo.toml]: since we are sticking with openssl, we should probably update the rest of our hyper based crates to using this.

**Comment by lukesteensen** [comment]
Is the failing test related?

**Comment by LucioFranco** [comment]
@lukesteensen yeah could be, ill take a look on monday

**Comment by binarylogic** [comment]
Before we merge, I'd like to add a simple "Custom DNS" section for all relevant sinks.

**Comment by LucioFranco** [comment]
@binarylogic would it make sense to just add a general DNS section to the global options page and users should just expect that the sinks etc are using that? Instead of adding a section to each sink/transform.

**Comment by binarylogic** [comment]
Yes, that works 👍

**Comment by LucioFranco** [comment]
@lukesteensen one way we might be able to consolidate is to attach some http client factory as a part of the `SinkContext`. Then all our sinks can use that or the resolver directly.

**Comment by LucioFranco** [comment]
@lukesteensen good catch, I totally forgot about that macro! That makes that much cleaner!

**Pull request merged**